Annotate System.Reflection.TypeExtensions for nullable references

danmoseley opened the pull request:
Contributes to #40623

jkotas committed:
Replace primitive Array fcalls with managed implementations (#27123)

Prep-work for #27106

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

AaronRobinsonMSFT committed:
Conditionally remove the GC transition from a P/Invoke (#26458)

* Provide mechanism to remove the GC transition from a P/Invoke.
  New Attribute: 'System.Runtime.InteropServices.SuppressGCTransitionAttribute'

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

stephentoub committed:
Use stackalloc expressions with builder ctors (dotnet/coreclr#27194)

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

GrabYourPitchforks committed:
Update Utf8String to match latest prototype (dotnet/coreclr#27046)

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

Commit:
Enable ETW/EventSource logging of task IDs for boxed state machines (#27115)

* Wrap MoveNext Action for TPL event tracing

Wraps the MoveNext action of the AsyncStateMachineBox in a
continuation wrapper when async causality tracing is on so that the
TPL event source can find the task that is associated with a
continuation. Does not wrap otherwise.

* Clarifying comment

* removing trailing whitespace

* code review feedback.

Makes AsyncMethodBuilderCore.TryFindContinuationTask check to
whether the target of a continuation is itself a task, as a fall-back
to checking to see if the continuation is a ContinuationWrapper

* Got rid of unnecessary null checks

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

GrabYourPitchforks committed:
Fix DecoderNLS.Convert to out the correct value for 'completed' (#27210)

This is a complementary fix to https://github.com/dotnet/coreclr/issues/23020

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

benaadams committed:
Improve Dictionary TryGetValue size/perfomance (dotnet/coreclr#27195)

* Dictionary avoid second bounds check in Get methods

* Add NullRef methods to Unsafe

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

jkotas committed:
Delete ENABLE_WINRT define (dotnet/coreclr#27240)

.NET Native specific code not needed anymore

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

danmoseley committed:
Annotate System.Reflection.TypeExtensions for nullable references

danmoseley committed:
Annotate System.Reflection.TypeExtensions for nullable references

safern commented on src/System.Reflection.TypeExtensions/ref/System.Reflection.TypeExtensions.cs:
```diff
@@ -15,12 +15,12 @@ public static partial class AssemblyExtensions
-        public static System.Reflection.MethodInfo GetRaiseMethod(this System.Reflection.EventInfo eventInfo, bool nonPublic) { throw null; }
-        public static System.Reflection.MethodInfo GetRemoveMethod(this System.Reflection.EventInfo eventInfo) { throw null; }
-        public static System.Reflection.MethodInfo GetRemoveMethod(this System.Reflection.EventInfo eventInfo, bool nonPublic) { throw null; }
+        public static System.Reflection.MethodInfo?  GetAddMethod(this System.Reflection.EventInfo eventInfo) { throw null; }
```
Nit, this has an extra space after the `?` -- did you do it manually or used `GenAPI`?

danmosemsft replied:
I forgot about GenAPI. Remind me how to do it? I ran `msbuild /t:GenerateReferenceAssemblySource` in both ref and src and it didn't seem to do it.

danmosemsft replied:
@buyaa-n ?

buyaa-n replied:
it should work from src folder, also the option is little different: 
`msbuild /t:GenerateReferenceSource`

safern replied:
Yeah that should do it.

stephentoub commented on src/System.Reflection.TypeExtensions/src/System/Reflection/TypeExtensions.cs:
```diff
@@ -177,25 +177,25 @@ public static PropertyInfo[] GetProperties(Type type, BindingFlags bindingAttr)
         }
 
-        public static PropertyInfo GetProperty(Type type, string name, Type returnType, Type[] types)
+        public static PropertyInfo? GetProperty(Type type, string name, Type returnType, Type[] types)
```
GitHub won't let me comment on it, but Type.IsAssignableFrom accepts a `Type?`... we don't want to do the same with the below IsAssignableFrom?

stephentoub replied:
Same for IsInstanceOfType

safern replied:
Ha, we commented at the same time 😄

safern commented on src/System.Reflection.TypeExtensions/src/System/Reflection/TypeExtensions.cs:
```diff
@@ -177,25 +177,25 @@ public static PropertyInfo[] GetProperties(Type type, BindingFlags bindingAttr)
         }
 
-        public static PropertyInfo GetProperty(Type type, string name, Type returnType)
+        public static PropertyInfo? GetProperty(Type type, string name, Type returnType)
```
It seems like `returnType` should be nullable. https://source.dot.net/#System.Private.CoreLib/shared/System/Type.cs,224

safern commented on src/System.Reflection.TypeExtensions/src/System/Reflection/TypeExtensions.cs:
```diff
@@ -177,25 +177,25 @@ public static PropertyInfo[] GetProperties(Type type, BindingFlags bindingAttr)
         }
 
-        public static PropertyInfo GetProperty(Type type, string name, Type returnType, Type[] types)
+        public static PropertyInfo? GetProperty(Type type, string name, Type returnType, Type[] types)
```
Ditto.

safern commented on src/System.Reflection.TypeExtensions/src/System/Reflection/TypeExtensions.cs:
```diff
@@ -177,25 +177,25 @@ public static PropertyInfo[] GetProperties(Type type, BindingFlags bindingAttr)
+        public static PropertyInfo? GetProperty(Type type, string name, Type returnType, Type[] types)
         {
             Requires.NotNull(type, nameof(type));
             return type.GetProperty(name, returnType, types);
```
GitHub won't let me comment but `IsAssignableFrom` accepts a null type for parameter c and returns false.

safern replied:
Also, `IsInstanceOfType` accepts a null `object`.

safern reviewed (approved):
Other than my comments, LGTM.

ahsonkhan committed:
Followup to add more extension data tests. (#41877)

tannergooding committed:
Improve codegen for IsNan (dotnet/coreclr#27272)

* Updating double.IsNaN and float.IsNaN to just `return x != x`

* Updating x86 codegen to optimize `x != x` for floating-point

* Changing the isnan optimization to only check the operand assigned registers

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

dotnet-maestro[bot] committed:
[master] Update dependencies from 4 repositories (#41850)

* Update dependencies from https://github.com/dotnet/arcade build 20191015.2

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19515.2
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19515.2
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19515.2
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19515.2
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19515.2
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19515.2
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19515.2
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19515.2
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19515.2
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19515.2
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19515.2
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19515.2
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19515.2
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19515.2

* Update dependencies from https://github.com/dotnet/core-setup build 20191016.19

- Microsoft.NETCore.App - 5.0.0-alpha1.19516.19
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19516.19
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19516.19

* Update dependencies from https://github.com/dotnet/corefx build 20191016.16

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19516.16
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19516.16

* Update dependencies from https://github.com/dotnet/standard build 20191016.1

- NETStandard.Library - 2.2.0-prerelease.19516.1

* Update dependencies from https://github.com/dotnet/arcade build 20191017.8

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19517.8
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19517.8
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19517.8
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19517.8
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19517.8
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19517.8
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19517.8
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19517.8
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19517.8
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19517.8
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19517.8
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19517.8
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19517.8
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19517.8

Marusyk committed:
Store Description value of ConfigurationProperty 2 (#41873)

* Store Description value of ConfigurationProperty

* Skip test on the .NET Framework

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/core-setup build 20191018.1 (#41892)

- Microsoft.NETCore.App - 5.0.0-alpha1.19518.1
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19518.1
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19518.1

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/coreclr build 20191017.4 (#41893)

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19517.4
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19517.4
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19517.4

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/corefx dotnet/standard (#41894)

* Update dependencies from https://github.com/dotnet/corefx build 20191018.1

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19518.1
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19518.1

* Update dependencies from https://github.com/dotnet/standard build 20191017.2

- NETStandard.Library - 2.2.0-prerelease.19517.2

steveharter committed:
Support IDictionary<string,TValue> without IDictionaryEnumerator

jkotas committed:
Rewrite Buffer.BlockCopy in C# (dotnet/coreclr#27216)

* Rewrite Buffer.BlockCopy in C#

Fixes #27106

* Workaround to enable type check optimizations for BlockCopy only

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

buyaa-n committed:
Annotate System.ObjectModel for nullable (#41808)

Annotate System.ObjectModel for nullable

ViktorHofer committed:
Disable MemoryCacheTest.Contains test (#41890)

vcsjones committed:
Discard unused locals cleanup (#41876)

* Discard result from GetC14NDigest.

GetC14NDigest is being used to update the `hashAlg` from the
`GetDigestedOutput`, as well as update the `_digestedSignedInfo` and
cache flag. The return value is not needed. Discard it to make that
clearer.

* Discard the result from `CertControlStore`.

The comment above the change explains the reason for discarding
the result.

* Ignore result from CryptAcquireContext.

The comment below justifies the reason for ignoring the local.

* Code review feedback.

layomia committed:
Populate non-immutable collections directly on deserialize (#41482)

* Populate non-immutable collections directly on deserialize

(rather than storing in temporary collections and using converters to
create and populate instances)

* Fix deserilizing nested dictionaries

* Don't get add method for types we can populate without reflection; fix failing tests

* Misc perf changes

* Additional changes

* More changes

* Address review feedback

* Address feedback

GrabYourPitchforks committed:
Add UTF8Encoding compat tests (#41910)

ViktorHofer committed:
Remove clean target and use arcade one (#41889)

joperezr committed:
Adding single statement if formatting rule to coding-style (#41906)

* Adding single statement if formatting rule to coding-style

cc: @stephentoub @carlossanlop 

This came up in a recent PR and I just wanted to make it more clear in our Coding guidelines.

* Apply suggestions from code review

Co-Authored-By: Sam Harwell <sam.harwell@microsoft.com>

ViktorHofer committed:
Prepare license and other files for the repo consolidation (#41912)

* Update .azure-ci.yml

* Rename LICENSE.TXT to LICENSE and others

dotnet-maestro[bot] committed:
[master] Update dependencies from 4 repositories (#41914)

* Update dependencies from https://github.com/dotnet/arcade build 20191018.2

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19518.2
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19518.2
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19518.2
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19518.2
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19518.2
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19518.2
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19518.2
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19518.2
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19518.2
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19518.2
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19518.2
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19518.2
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19518.2
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19518.2

* Update dependencies from https://github.com/dotnet/core-setup build 20191018.11

- Microsoft.NETCore.App - 5.0.0-alpha1.19518.11
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19518.11
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19518.11

* Update dependencies from https://github.com/dotnet/corefx build 20191018.10

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19518.10
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19518.10

* Update dependencies from https://github.com/dotnet/standard build 20191018.1

- NETStandard.Library - 2.2.0-prerelease.19518.1

ViktorHofer committed:
Use Loopback for PreAuthenticate request test (#41891)

* Use Loopback for PreAuthenticate request test

* Remove outgoing web request

danmoseley committed:
Annotate System.Resources.ResourceWriter for nullable ref types (#41880)

* Annotate System.Resources.ResourceWriter for nullable ref types

* ref

* Mark common as nullable enable explicitly

* _resourceList

filipnavara committed:
Fix invalid IL generated in Xml serializers (#39765)

shortspider committed:
Improve the performance of ImmutableList.Contains (#40540)

* Improve ImmutableList.IndexOf performance

* Implement a spicific Contains method

IndexOf has a lot of overhead because it's looking for an index when
all Contains wants to do is find a matching element.

* Fix tests

* Revert change to IndexOf

VSadov committed:
A few more test scenarios for casting of nullables and open generic arrays. (#41918)

* A few more test scenarios for casting of nullables and open generic arrays.

filipnavara committed:
Move portable thread pool to shared partition (dotnet/corert#7828)

* Move portable thread pool to shared partition

* Rename ClrThreadPool to PortableThreadPool

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/core-setup build 20191019.8 (#41925)

- Microsoft.NETCore.App - 5.0.0-alpha1.19519.8
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19519.8
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19519.8

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/corefx dotnet/standard (#41927)

* Update dependencies from https://github.com/dotnet/corefx build 20191019.7

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19519.7
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19519.7

* Update dependencies from https://github.com/dotnet/standard build 20191019.1

- NETStandard.Library - 2.2.0-prerelease.19519.1

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/coreclr (#41926)

* Update dependencies from https://github.com/dotnet/coreclr build 20191019.1

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19519.1
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19519.1
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19519.1

* Update argument names for CoreLib change

idlerboris committed:
Add tests to improving TimeSpan precision (#41499)

Fix #41380

tmds committed:
StreamPipeWriter: don't flush when the writer is completed with an exception. (#40874)

This provides a way to discard buffered data without pushing it to the inner Stream.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/core-setup build 20191021.2 (#41944)

- Microsoft.NETCore.App - 5.0.0-alpha1.19521.2
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19521.2
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19521.2

ViktorHofer committed:
Disabling TaskCanceledQuickly timing out test (#41941)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/coreclr build 20191020.2 (#41945)

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19520.2
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19520.2
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19520.2

ViktorHofer committed:
Update netfx queue to Windows 10 19H1 (#41942)

danmoseley committed:
Fix XSLT tests to deterministically pick newer xsltc.exe (#41919)

* Fix XSLT tests

* Typo

* Typo

* Nits

carlossanlop committed:
Backup important wiki information (#41774)

* Backup important wiki information

* Restore original title in unix-instructions.md

* FreeBSD instructions only for 3.0

* Remove conferences.md

* update unix-instructions.md title

* Move FreeBSD link to the end of unix-instructions.md

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/corefx dotnet/standard (#41948)

* Update dependencies from https://github.com/dotnet/corefx build 20191020.7

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19520.7
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19520.7

* Update dependencies from https://github.com/dotnet/standard build 20191020.1

- NETStandard.Library - 2.2.0-prerelease.19520.1

MichalStrehovsky committed:
Update breaking change rules for default interface methods (#41949)

* Update breaking change rules for default interface methods

* Apply suggestions from code review

Co-Authored-By: Stephen Toub <stoub@microsoft.com>

Marusyk committed:
Remove dead code null check from DirectoryEntry (#41843)

* Fix possible NRE in DirectoryEntry from System.DirectoryServices

* Update DirectoryEntry.cs

clamp03 committed:
Fix a build error on ARM (#41937)

- Add type casting in CryptoNative_X509VerifyCertErrorString

sonnemaf committed:
Add readonly to members of System.Drawing.Primitives (#40407)

jkotas committed:
Cleanup memory copy helpers (dotnet/coreclr#27307)

- Merge RuntimeImports that has just a few FCalls into Buffer.
- Delete assembly implementation of memcpy for ARM. The implementation was outdated and it was used only on Windows ARM and only for fraction of memcpy callsites.
- Delete unnecessary PInvoke into SysStringLen. We have a managed equivalent on System.Runtime.InteropServices.Marshal.
- Delete large amount of redundant code between SecureString.Windows.cs and SecureString.Unix.cs, and switch SecureString to use Spans
- Fold null check into IsWin32Atom

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

stephentoub committed:
Don't root FileSystemWatcher unnecessarily (#41872)

We already try to do this on Unix, though it seems we don't currently have tests for it, but we don't on Windows.

madmir committed:
Refactor DllImports in System.Reflection.Metadata (#40579)

svick committed:
Cache polymorphic properties (#41753)

* Cache polymorphic properties

* Move RuntimePropertyCache to JsonClassInfo

* Added test of RuntimePropertyCache using properties with different attributes

* Fixed typo

Co-Authored-By: Ahson Khan <ahkha@microsoft.com>

* Use allocating overload of GetOrAdd on .Net Standard 2.0

safern committed:
Update readme badge to point to public CI build (#41970)

stephentoub committed:
Guard against attempting to creating out-of-range spans in macOS FileSystemWatcher (#41957)

* Guard against attempting to creating out-of-range spans in macOS FilesystemWatcher

Fixes https://github.com/dotnet/corefx/issues/40054

* Address PR feedback

steveharter committed:
Merge branch 'master' of https://github.com/steveharter/dotnet_corefx into DictionaryOfT

steveharter committed:
Additional merge issues and non-generic support for that

adamsitnik committed:
remove double bound check from StringBuilder.Append(char) (#27340)

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

jkotas committed:
Delete Uap quirks from CoreLib (dotnet/coreclr#27356)

This matches cleanup done in CoreFX.

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

ahsonkhan committed:
Use Sse2 instrinsics to make NeedsEscaping check faster for large JSON strings (#41845)

* Use Sse2 instrinsics to make NeedsEscaping check faster for large
strings.

* Update the utf-8 bytes needsescaping and add tests.

* Remove unnecessary bitwise OR and add more tests

* Add more tests around surrogates, invalid strings, and characters >
short.MaxValue.

ViktorHofer committed:
Fix clean switch (#41982)

ViktorHofer committed:
Disable ComponentModel MemberDescriptor parallel (#41985)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20191021.4 (#41987)

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19521.4
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19521.4
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19521.4
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19521.4
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19521.4
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19521.4
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19521.4
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19521.4
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19521.4
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19521.4
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19521.4
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19521.4
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19521.4
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19521.4

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/coreclr build 20191021.1 (#41988)

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19521.1
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19521.1
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19521.1

vcsjones committed:
Cleanup unused locals in more places (#41964)

* Removed unused locals from AsnReader.

The return value of the method is not needed however the out values
are.

* Remove key local from ECDiffieHellmanCng.

The return value is not needed however the key still needs to be
generated.

* Remove key local from ECDsaCng.

The return value is not needed however the key still needs to be
generated.

* Remove unused local.

The return value is not needed from the method however the out
parameters are.

* Discard result from `CryptAcquireContext`.

The result is discarded rather than entirely omitted to clearly express
the intent that the return value is not used.

stephentoub committed:
Fix a rare GetAwaiter().GetResult() on a ValueTask in System.Net.Http (#41958)

* Fix a rare GetAwaiter().GetResult() on a ValueTask in System.Net.Http

If someone's custom HttpContent-derived type is implementing their async SerializeToStreamAsync with synchronous Write calls on the destination stream, and if that WriteAsync method returns an IValueTaskSource-based ValueTask that's not yet completed, this could throw an exception.   This is unlikely to actually negative impact anyone right now, but it's worth fixing.  This is one of the patterns an analyzer around ValueTasks will help find and fix.

* Update src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpBaseStream.cs

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/core-setup dotnet/corefx dotnet/standard (#41990)

* Update dependencies from https://github.com/dotnet/core-setup build 20191021.12

- Microsoft.NETCore.App - 5.0.0-alpha1.19521.12
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19521.12
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19521.12

* Update dependencies from https://github.com/dotnet/corefx build 20191021.4

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19521.4
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19521.4

* Update dependencies from https://github.com/dotnet/standard build 20191021.1

- NETStandard.Library - 2.2.0-prerelease.19521.1

steveharter committed:
Merge pull request #41903 from steveharter/DictionaryOfT

Support IDictionary<string,TValue> without IDictionaryEnumerator

filipnavara committed:
Fix nullability error in portable thread pool (dotnet/coreclr#27366)

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

eiriktsarpalis committed:
Add windows support to build.sh (#41947)

tannergooding committed:
Adding back more arm intrinsics that existing before refactoring (#27153)

* Adding AdvSimd.And

* Adding AdvSimd.AndNot

* Adding AdvSimd.Or

* Adding AdvSimd.OrNot

* Adding AdvSimd.Xor

* Adding AdvSimd.Not

* Removing a trailing whitespace from AdvSimd.PlatformNotSupported.cs

* Adding AdvSimd.Subtract

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

safern committed:
Update TsaBugAreaPath to fix SDL Validation (#41993)

stephentoub committed:
Annotate System.Runtime.Serialization.Primitives for nullable reference types (#41992)

carlossanlop committed:
Unit tests to verify value returned by DirectoryInfo.Parent.ToString, both in CoreFX and in Framework. (#41452)

ViktorHofer committed:
Rename LICENSE to LICENSE.TXT (#42000)

AntonLandor committed:
System.IO.FileSystem.MoveFile(...) will now delete source file after copying it to its destination when moving files over file systems. (#41767)

ViktorHofer committed:
Revert license renaming (#42002)

JeremyKuhne committed:
Handle non blittable LOGFONT structs (#41975)

* Handle non blittable LOGFONT structs

Adding support for blittable structs broke non-blittable structs passed to `Font.FromLogFont`. Try to blit and fall through to the marshaller for non blittable types.

Fixes: https://github.com/dotnet/winforms/issues/2136

* Clarify comment block

stephentoub committed:
Annotate System.Text.Encoding.CodePages for nullable reference types (#41991)

* Annotate System.Text.Encoding.CodePages for nullable reference types

* Fix .NET Core 2.0 build

I'm not sure why we still build for this as a target, but there a bunch of nullability failures in the build because Debug.Assert isn't annotated in .NET Core 2.0's surface area, and thus places we're relying on Debug.Assert to suppress nullable warnings are failing.

AaronRobinsonMSFT committed:
Expose SuppressGCTransitionAttribute (#41997)

Dotnet-GitSync-Bot committed:
Add documentation for SuppressGCTransition (dotnet/coreclr#27368) (#42006)

Signed-off-by: arobins@microsoft.com

GrabYourPitchforks committed:
Add Encoding extensions which work with sequences (#41810)

tannergooding committed:
Adding Unsafe.SkipInit (#41995)

* Adding Unsafe.SkipInit

* Removing the "where T : struct" constraint from Unsafe.SkipInit

* Adding tests for Unsafe.SkipInit

tannergooding committed:
Move Vector2/3/4 to the shared CoreLib partition (#42005)

ahsonkhan committed:
Optimize FindFirstCharToEncode for JavaScriptEncoder.Default and Relaxed using Sse2 intrinsics. (#41933)

* Optimize FindFirstCharToEncode for JavaScriptEncoder.Default and Relaxed
using Sse2 intrinsics.

* Create an Sse2Helper and improve perf of TextEncoder and
AllowedCharactersBitmap

* Loop unroll FindFirstCharacterToEncode

* Improve code coverage.

* Add more tests for surrogate pairs and fix call to WillEncode.

* Address PR feedback - remove some code duplication.

* Move DefaultJavaScriptEncoder to separate file and override EncodeUtf8
with better caching.

* Add default replacement character as a test.

* Address nits.

tmds committed:
Socket.Select: increase ref count while the handle is in use (#41763)

* Socket.Select: increase ref count while the handle is in use

* PR feedback

* Use Socket.InternalSafeHandle for ref/release.

* Socket.Windows: SelectFileDescriptor: add missing unrefs

tannergooding committed:
Regenerating the ref assembly for System.Runtime.Intrinsics.Experimental

jkotas committed:
Simplify LOGFONT marshalling (#42011)

carlossanlop committed:
Add directory creation method that takes an ACL (#41834)

* First commit

* Fix OperationCanceledException dependency error, revert nullable indication

* Only Directory files

* Unit tests verifications

* Split FileSystem.Windows into two additional files instead of one. Fix pinning.

* Remove dead comment, add actual UT verifications

* User WellKnownSidType

* Make method unsafe, nit comments addressed.

* Remove irrelevant comment

* Add additional exceptions to triple slash

* Add unit tests for additional exceptions

* Remove FileSystem.cs, it has one obsolete method. Replace all calls of that removed method.

* Second pass on required strings.resx resources

* Remove unused dependency cs files

* Move shared files to Common

* Add the missing braces

* Add nulcheck for this argument, add exception, document exception in triple slash, add unit test.

* Fix netstandard CI build failure

* Fix Microsoft.IO.Redist build issue with NetStandard

* Remove unnecessary assert in ut

* Fix build problem in Windows Packaging All Configurations x64_Debug, System.Runtime needs to be included to recognize NotNullIfNotNullAttribute, consumed in Path.Internal.cs

* Additional file splitting and dependency addition to fix netstandard/netfx build problems

* Fix netstandard and netcoreapp build issues

* Remove unused file

* nit cleaning

* Minimal unit tests

* Remove unused unit test code and spacing fix

* Add full framework condition to unit test

* Additional fix for full framework

stephentoub committed:
Fix several Socket tests to not dispose an event (#42020)

When I run these tests locally, due to my network configuration they sometimes fail.  If they fail and the test is disposing of the ManualResetEvent, the async callback may try to set the event after it's already been disposed, which throws an exception that crashes the test process.  The easy fix is to just dispose of these MRE instances on success and let their finalizers clean up for them if the tests fail.

tannergooding committed:
Adding SkipInit to Internal.Runtime.CompilerServices.Unsafe (#27377)

* Adding SkipInit to Internal.Runtime.CompilerServices.Unsafe

* Updating the runtime to handle Unsafe.SkipInit

* Updating Decimal.DecCalc to use Unsafe.SkipInit

* Removing 'workaround for CS0165' comments

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

AaronRobinsonMSFT committed:
Apply SuppressGCAttribute to some SPCL functions. (dotnet/coreclr#27369)

Apply SuppressGCAttribute to some SPCL functions.

Only insert GC_POLL in first morph call.

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

ahsonkhan committed:
JsonSerializer.Serialize should check whether the passed-in (#42026)

Utf8JsonWriter is null.

ahsonkhan committed:
When encoder is null, use JavaScriptEncoder.Default to check for NeedsEscaping (#42023)

* When encoder is null, use JavaScriptEncoder.Default to check for
NeedsEscaping.

* Remove unnecessary unsafe keyword and add comment to using directive.

* Address feedback.

* Remove gotos and move the IsEmpty check outside the fixed block.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/coreclr build 20191022.3

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19522.3
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19522.3
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19522.3

tannergooding committed:
Move Matrix3x2/4x4, Plan, and Quaternion to the shared CoreLib partition (#42021)

stephentoub committed:
Add test for SafeHandle handling in Socket.Select (#42018)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20191022.8 (#42031)

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19522.8
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19522.8
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19522.8
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19522.8
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19522.8
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19522.8
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19522.8
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19522.8
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19522.8
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19522.8
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19522.8
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19522.8
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19522.8
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19522.8

eriawan committed:
Remove unused locals in System.Data.Common and keep some logic using discard (#41962)

* remove unused locals in System.Data.Common and keep some logic using discard

* revert the HandleTable(element) removal as PR feedback. Use discard as return value is ignored.

danmoseley committed:
Event log retries (#42017)

tmds committed:
Remove Socket.InnerSafeCloseSocket (#41888)

* Remove Socket.InnerSafeCloseSocket

* Move IsInvalid checks in SetHandle function

* PR feedback

* Remove unnecessary 'SocketPal.'

* Remove SafeSocketHandle.SetHandle

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/core-setup dotnet/corefx dotnet/standard (#42036)

* Update dependencies from https://github.com/dotnet/core-setup build 20191022.10

- Microsoft.NETCore.App - 5.0.0-alpha1.19522.10
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19522.10
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19522.10

* Update dependencies from https://github.com/dotnet/corefx build 20191023.1

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19523.1
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19523.1

* Update dependencies from https://github.com/dotnet/standard build 20191022.1

- NETStandard.Library - 2.2.0-prerelease.19522.1

stephentoub committed:
Delete duplicative exception add in SourceCore (#42035)

marek-safar committed:
Remove unused methods (dotnet/coreclr#27370)

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

grisha-kotler committed:
Reduce the number of syscalls when converting Ticks to TimeSpan (#41989)

* Reduce the number of syscalls when converting Ticks to TimeSpan

* Remove the usage of Lazy

* Fix build

timandy committed:
Optimize Enumerable.SkipLast() for IPartition and IList (#41342)

* Optimize Enumerable.SkipLast() for IPartition and IList

* Convert to conditional expression

* Add parameter name

* Revert "Convert to conditional expression"

This reverts commit 7c7908fb

* Format conditional expression

stephentoub committed:
Enable pooling for `async ValueTask/ValueTask<T>` methods (#26310)

Today `async ValueTask/ValueTask<T>` methods use builders that special-case the synchronously completing case (to just return a `default(ValueTask)` or `new ValueTask<T>(result))` but defer to the equivalent of `async Task/Task<T>` for when the operation completes asynchronously. This, however, doesn't take advantage of the fact that value tasks can wrap arbitrary `IValueTaskSource/IValueTaskSource<T>` implementations.

This commit gives `AsyncValueTaskMethodBuilder` and `AsyncValueTaskMethodBuilder<T>` the ability to use pooled `IValueTaskSource/IValueTaskSource<T>` instances, such that calls to an `async ValueTask/ValueTask<T>` method incur 0 allocations (ammortized) as long as there's a cached object available.

Currently the pooling is behind a feature flag, requiring opt-in via the DOTNET_SYSTEM_THREADING_POOLASYNCVALUETASKS environment variable (setting it to "true" or "1").  This is done for a few reasons:
- There's a breaking change here, in that while we say/document that `ValueTask/ValueTask<T>`s are more limited in what they can be used for, nothing in the implementation actually stops a `ValueTask` that was wrapping a `Task` from being used as permissively as `Task`, e.g. if you were to `await` such a `ValueTask` multiple times, it would happen to work, even though we say "never do that". This change means that anyone who was relying on such undocumented behaviors will now be broken. I think this is a reasonable thing to do in a major release, but I also want feedback and a lot of runway on it.
- Policy around pooling. Pooling is always a difficult thing to tune. Right now I've chosen a policy that limits the number of pooled objects per state machine type to an arbitrary multiple of the processor count, and that tries to strike a balance between contention and garbage by using a spin lock and if there's any contention while trying to get or return a pooled object, the cache is ignored. We will need to think hard about what policy to use here. It's also possible it could be tuned per state machine, e.g. by having an attribute that's looked up via reflection when creating the cache for a state machine, but that'll add a lot of first-access overhead to any `async ValueTask/ValueTask<T>` method.  For now, it's tunable via the `DOTNET_SYSTEM_THREADING_POOLASYNCVALUETASKSLIMIT` environment variable, which may be set to the maximum number of pooled objects desired per state machine.
- Performance validation.  Preliminary numbers show that this accomplishes its goal, having on-par throughput with the current implementation but with significantly less allocation.  That needs to be validated at scale and across a variety of workloads.
- Diagnostics.  There are several diagnostics-related abilities available for `async Task/Task<T>` methods that are not included here when using `async ValueTask/ValueTask<T>` when pooling.  We need to evaluate these (e.g. tracing) and determine which pieces need to be enabled and which we're fine omitting.

Before shipping .NET 5, we could choose to flip the polarity of the switch (making it opt-out rather than opt-in), remove the fallback altogether and just have it be always on, or revert this change, all based on experimentation and data we receive between now and then.

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

Marusyk committed:
Recombine files ProcessTests.cs and ProcessTests.netcoreapp.cs (#42063)

* Recombine files ProcessTests.cs and ProcessTests.netcoreapp.cs that aren't supported anymore

* Add namespaces

echesakov committed:
Add AdvSimd (LeadingZeroCount, LeadingSignCount, PopCount) (#27382)

* AdvSimd.LeadingSignCount

* AdvSimd.LeadingZeroCount

* AdvSimd.PopCount

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

omajid committed:
Add cgroup v2 support to Interop.cgroups (#39686)

* Add cgroup v2 support to Interop.cgroups

Fix up code to adjust cgroup v1 assumptions and check cgroup v2 paths,
locations and values.

Continue using the older cgroup v1 terminology for APIs.

* Clean up new tests

VSadov committed:
Moving StopWatch to `src\Common\src\Corelib` (#42072)

bartonjs committed:
Normalize Linux and Windows X.509 Revocation Statuses (#41956)

* Add CrlCrl test before revocation support

* Add revocation tests

* Add lots of tests with revocation

* Make Linux download respect time remaining

* Use synchronous wait so Linux gets an answer

* Use IPv4Loopback name instead of localhost for stability (IPv4 vs IPv6)

* Some tracing improvements

* Add a lot more tests

* Linux multi-OCSP and revocation normalization support

* Even more tests, mainly invalid responses

* Respect CorruptRevocationIssuerName

* Fix the "no timeout" overflow.

* Treat OfflineRevocation as always suppressed

* Finish some half-finished tests

* De-spaghetti Linux OCSP processing

* Check what happens for testing the root directly

* Native changes for Linux multi-layer OCSP

* Make Linux portable build work

* Mitigate issue 41969 in tests

* Support decoding incorrect encoding of OCSPRequest

Also fix an exact-tag vs loose-tag check.

* Disable DynamicRevocationTests on macOS

* Disable designated responder tests on RHEL6.

* Cache the error string for each error code

* Self-review cleanup

* Disable designated OCSP responder tests on Win7

* Re-enable WIn7 designated responder tests, increase timeouts.

* Running locally on Win7 the designated responder tests pass
* Observation of the intermittent CI failures shows they were always in excess of 5 seconds.
  * Running locally the longest test was 1 second, and that includes generating several RSA keys.

* Apply review feedback

* Move DynamicRevocationTests to OuterLoop

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20191023.3 (#42080)

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19523.3
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19523.3
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19523.3
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19523.3
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19523.3
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19523.3
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19523.3
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19523.3
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19523.3
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19523.3
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19523.3
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19523.3
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19523.3
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19523.3

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/coreclr build 20191023.1 (#42081)

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19523.1
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19523.1
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19523.1

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/core-setup dotnet/corefx dotnet/standard (#42082)

* Update dependencies from https://github.com/dotnet/core-setup build 20191023.10

- Microsoft.NETCore.App - 5.0.0-alpha1.19523.10
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19523.10
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19523.10

* Update dependencies from https://github.com/dotnet/corefx build 20191023.8

- runtime.native.System.IO.Ports - 5.0.0-alpha1.19523.8
- Microsoft.NETCore.Platforms - 5.0.0-alpha1.19523.8

* Update dependencies from https://github.com/dotnet/standard build 20191023.1

- NETStandard.Library - 2.2.0-prerelease.19523.1

jkotas committed:
Fix build break

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

tannergooding committed:
Move down S.N.Vector APIs and expose new conversion support (#27401)

* Updating S.P.Corelib to start building Vector2/3/4 and supporting types

* Removing HAS_INTRINSICS usages from Matrix4x4

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

adamsitnik committed:
Optimize GC.AllocateUninitializedArray and use it in StringBuilder (#27364)

* use GC.AllocateUninitializedArray for allocating internal char buffers

* force inlining of AllocateUninitializedArray to have no perf hit on the small buffers hot path

* insrease the threshold from 256 to 2048 bytes

* use Unsafe.As instead of a cast

* remove the size precondition from AllocateNewArray method, the called AllocateSzArray is responsible for handling negative size

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

wfurt committed:
add entries for new FreeBSD releases (#41794)

stephentoub committed:
Fix non-netcoreapp builds

layomia committed:
Ensure serializer throws JsonException when deserializing incompatible JSON into collections (#42066)

* Ensure serializer throws JsonException when deserializing incompatible JSON into collections

* Address review feedback

tarekgh committed:
Fix RegionInfo.DisplayName test (#42088)

noahfalk committed:
DiagnosticSourceEventSource fixes/improvements for distributed tracing (#41943)

* DiagnosticSourceEventSource fixes for distributed tracing

1) There is no existing mechanism for DiagnosticSourceEventSource to fetch the current activity object. By convention it is not passed as an event argument, but rather stored in the async-local property Activity.Current. This was fixed by adding a well-known "*Activity" property that returns the result of Activity.Current regardless what object it is applied to.

2) DiagnosticSourceEventSource fails to evaluate properties on value types, such as DateTime.Ticks. Calling MethodInfo.CreateDelegate
needs to use a different signature for ref and value type properties and previously the code always used the ref-style signature. Fixed by adding ValueTypedFetchProperty that does the proper CreateDelegate and delegate invocation for structs.

3) There is no mechanism for DiagnosticSourceEventSource to enumerate the tags on an activity. This change adds the *Enumerate well-known property which will iterate any IEnumerable`1, invoke ToString() on each element, then join it as a string.

sbomer committed:
Update ILLink.Tasks (#39287)

* Update ILLink.Tasks

* Don't import linker nuget package targets

We use custom targets to invoke the linker.

* Don't pass directories as assembly paths

Instead pass them as extra linker args. To avoid issues with the
response file parsing logic, we remove trailing slashes.

* Update to linker version with unusedinterfaces opt

* Update System.Text.Json with more ctor roots

bartonjs committed:
Clean up some X509FilesystemTests tests (#42089)

* Make all of the Fact methods public
* Improve the assert messages in VerifyCrlCache
* Make VerifyCrlCache understand offline revocation unknown asserts two flags now.

mmitche committed:
Switch to label.N form for pre-release label (#42090)

In order to facilitate better preview sorting, switch to label.N form for the pre-release label.

wfurt committed:
improve performance of GetAllNetworkInterfaces on Linux (#41490)

* improve GetLinuxNetworkInterfaces

* more improvements to GetAllNetworkInterfaces()

* fix compilation

* update ethtool to support older kernels

* fix osx

* more platform fixes

* put PrefixLengthToSubnetMask back as it was

* normalize units properlry

* fix constant

* feedback from review

* fix log initialization

krwq committed:
Support PinChangedEvent on Unix (#40383)

* Support PinChangedEvent on Unix

* address PR feedback

* Fix typo in RaisePinChanged

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/coreclr (#42094)

* Update dependencies from https://github.com/dotnet/coreclr build 20191024.2

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19524.2
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19524.2
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19524.2

* Update ref/System.Runtime.Intrinsics.Experimental.cs corresponding to https://github.com/dotnet/coreclr/pull/27382/

* Updating System.Numerics.Vectors to type forward to S.P.Corelib

* Regenerating the System.Runtime.Intrinsics ref assembly for existing APIs

* Regenerating the System.Runtime.Intrinsics ref assembly for new APIs

* Removing DefaultTargets="Build" from the S.R.I and S.N.V Configurations.props

alnikola committed:
Use ValueStringBuilder in UriHelper.UnescapeString (#41684)

* UriExt.UnescapeDataString avoids a new string allocation for an unmodified unescaped string

* Original dest array reuse fixed

* ValueStringBuilder is used instead of PooledCharArray

* Remove unused type

* Changes to ValueStringBuilder reverted

* IsStringSame replaced with SequenceEqual

* Array returned from buffer is cleared

* - Indexer-based assignment is replaced with Append method
- Explicit pooled array cleaning is removed

* .sln file changes reverted

* Build fixed

* UnescapeString always copies the result to dest array

* Unused code removed

* Original string reuse fixed

* Heap validation fixed

* - Small optimizations
- Code style fixes

* By-char append is replaced with span-based one

* SequenceEqual is invoked against the original string

* Revert "By-char append is replaced with span-based one"

This reverts commit e5919e3dbe4be0681540c5084fce23d026149957.

eugenesmlv committed:
Add tests to verify that JSON deserializer doesn't do any multiplication or addition on the string length (#41754)

* Add deserialization tests for very long input strings (#41687)

* Fix tests

* Add deserialization tests for the maximum possible string length

* Improve long strings creation in deserialization tests for netcoreapp

* Extract common logic from if/def block

* Fix assertion

* Remove unnecessary test cases

* Remove other extra test cases

eiriktsarpalis committed:
HttpStress: Add Stress Pipelines (#41833)

safern committed:
Update SDL validation to use values from corefx-sdl-validation variable group (#42098)

saucecontrol committed:
fix license link (#42103)

ahsonkhan committed:
Update the S.T.E.W configurations to explicitly target a versioned TFM (nc3.0). (#42069)

* Update the S.T.E.W configurations to explicitly target a versioned TFM
(nc3.0).

* Use custom defined constant like S.T.Json - Building_Inbox_Library

* Update config.props to use package config and leave netcoreapp as a build config

* Add back netcoreapp specific config to the src csproj

* Revert "Use custom defined constant like S.T.Json - Building_Inbox_Library"

This reverts commit 8648cf6d7f5cbe54a6b1facdebc3f33c255572a1.

* Remove extra new line.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20191024.2 (#42110)

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19524.2
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19524.2
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19524.2
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19524.2
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19524.2
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19524.2
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19524.2
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19524.2
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19524.2
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19524.2
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19524.2
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19524.2
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19524.2
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19524.2

Youssef1313 committed:
Use NameOf in Conversion.vb (#42117)

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/core-setup dotnet/corefx dotnet/standard (#42115)

* Update dependencies from https://github.com/dotnet/core-setup build 20191024.4

- Microsoft.NETCore.App - 5.0.0-alpha1.19524.4
- Microsoft.NETCore.DotNetHostPolicy - 5.0.0-alpha1.19524.4
- Microsoft.NETCore.DotNetHost - 5.0.0-alpha1.19524.4

* Update dependencies from https://github.com/dotnet/corefx build 20191024.14

- runtime.native.System.IO.Ports - 5.0.0-alpha.1.19524.14
- Microsoft.NETCore.Platforms - 5.0.0-alpha.1.19524.14

* Update dependencies from https://github.com/dotnet/standard build 20191024.1

- NETStandard.Library - 2.2.0-prerelease.19524.1

buyaa-n committed:
Annotate System.Linq.Parallel for nullable ref types (#41533)

* Nullable annotation for System.Linq.Parallel

stephentoub committed:
Remove ILLinkTrim.xml from System.Text.Json (#42118)

* Remove ILLinkTrim.xml from System.Text.Json

Instead use PreserveDependencyAttribute.

* Address PR feedback

yukiwongky committed:
Fix getting readonly column metadata

safern committed:
Add support to produce WebAssembly runtime assets (#41777)

* Add support to produce WebAssembly runtime assets

* Add support to build in OSX and address PR feedback

* Change AND -> and in Directory.Build.props change region

* PR Feedback: run build in rolling CI builds as well

* Add TODO comment for WebAssembly OSGroup

buyaa-n committed:
Annotate System.Reflection.DispatchProxy for nullable (#41904)

* Nullable annotation for System.IO.Compression

* Revert "Nullable annotation for System.IO.Compression"

* Annotate System.Reflection.DispatchProxy for nullable

* Applying feedback

* Applying feedback

* Banging some null assertions causing error on CI

* Banging some more to clean CI build

safern committed:
Revert "Add support to produce WebAssembly runtime assets (#41777)" (#42127)

This reverts commit c57cba795a0e0a2905f6ce48009cde1d23cd7a19.

yukiwongky committed:
Merge pull request #42122 from yukiwongky/readonlycol

Fix getting readonly column metadata

stephentoub committed:
Use ValueStringBuilder instead of StringBuidler{Cache} in multiple places

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

stephentoub committed:
Avoid unnecessary allocations in Exception.ToString

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

dotnet-maestro[bot] committed:
[master] Update dependencies from dotnet/coreclr (#42121)

* Update dependencies from https://github.com/dotnet/coreclr build 20191025.1

- Microsoft.NET.Sdk.IL - 5.0.0-alpha1.19525.1
- Microsoft.NETCore.ILAsm - 5.0.0-alpha1.19525.1
- Microsoft.NETCore.Runtime.CoreCLR - 5.0.0-alpha1.19525.1

* remove Stopwatch.*.cs from System.Runtime.Extensions build

* Disable tests against CoreCLR bug

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20191025.2 (#42137)

- Microsoft.DotNet.XUnitExtensions - 5.0.0-beta.19525.2
- Microsoft.DotNet.XUnitConsoleRunner - 2.5.1-beta.19525.2
- Microsoft.DotNet.VersionTools.Tasks - 5.0.0-beta.19525.2
- Microsoft.DotNet.ApiCompat - 5.0.0-beta.19525.2
- Microsoft.DotNet.Arcade.Sdk - 5.0.0-beta.19525.2
- Microsoft.DotNet.Build.Tasks.Configuration - 5.0.0-beta.19525.2
- Microsoft.DotNet.Build.Tasks.Feed - 5.0.0-beta.19525.2
- Microsoft.DotNet.Build.Tasks.Packaging - 5.0.0-beta.19525.2
- Microsoft.DotNet.CodeAnalysis - 5.0.0-beta.19525.2
- Microsoft.DotNet.CoreFxTesting - 5.0.0-beta.19525.2
- Microsoft.DotNet.GenAPI - 5.0.0-beta.19525.2
- Microsoft.DotNet.Helix.Sdk - 5.0.0-beta.19525.2
- Microsoft.DotNet.RemoteExecutor - 5.0.0-beta.19525.2
- Microsoft.DotNet.GenFacades - 5.0.0-beta.19525.2

benaadams committed:
Avoid mod operator when fast alternative available (dotnet/coreclr#27299)

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

jkotas committed:
Delete a few more instances of StringBuilderCache (#42139)

jkotas committed:
Fix build break

stephentoub committed:
Remove ILLinkTrim.xml from System.Net.Http (#42126)

The file was there to keep anonymous types used by DiagnosticsHandler from having their getters removed.  But a) it was relying on the implementation detail name generated by Roslyn, and b) it was keeping alive other state from other anonymous types in the assembly.  I changed it to use concrete types instead.  This also happens to result in less code, because we don't need the generated Equals/GetHashCode methods generated for anonymous types.

stephentoub committed:
Clean up use of PreserveDependencyAttribute in System.Text.Json (#42146)

It's not necessary to specify the assembly name when the consumer is in the same assembly.

dotnet-maestro[bot] committed:
Update dependencies from dotnet/corefx (dotnet/coreclr#27431)

* Update dependencies from https://github.com/dotnet/corefx build 20191024.13

- Microsoft.NETCore.Platforms - 5.0.0-alpha.1.19524.13
- Microsoft.Private.CoreFx.NETCoreApp - 5.0.0-alpha.1.19524.13

* Update dependencies from https://github.com/dotnet/corefx build 20191024.14

- Microsoft.NETCore.Platforms - 5.0.0-alpha.1.19524.14
- Microsoft.Private.CoreFx.NETCoreApp - 5.0.0-alpha.1.19524.14

* Update dependencies from https://github.com/dotnet/corefx build 20191025.6

- Microsoft.NETCore.Platforms - 5.0.0-alpha.1.19525.6
- Microsoft.Private.CoreFx.NETCoreApp - 5.0.0-alpha.1.19525.6

* Workaround https://github.com/dotnet/coreclr/issues/27465

Signed-off-by: dotnet-bot <dotnet-bot@microsoft.com>

stephentoub committed:
Remove ILLinkTrim.xml from System.Diagnostics.DiagnosticSource (#42124)

- Use PreserveDependencyAttribute for ctors used via reflection by FetcherForProperty
- Remove reflection use entirely when looking up Activity events

jzabroski committed:
Update public-signing.md (#42129)

@jaredpar Updated this doc based on comments on Twitter discussion with you and @dsplaisted

ViktorHofer committed:
Remove versionless netstandard target group (#41984)

danmoseley committed:
CR

danmosemsft commented:
I do not understand why it is showing all these commits, when my branch is rebased and looks normal:
https://github.com/dotnet/corefx/compare/master...danmosemsft:S.R.TE
When you go there it shows this PR!
I'll open a new one.
